azure-storage/src/main/scala/akka/stream/alpakka/azure/storage/impl/AzureStorageStream.scala

```diff
@@ -0,0 +1,282 @@
+  ClearFileRange,
+  CreateContainer,
+  CreateDirectory,
+  CreateFile,
+  DeleteContainer,
+  DeleteDirectory,
+  GetProperties,
+  RequestBuilder,
+  UpdateFileRange
+}
+import akka.stream.scaladsl.{Flow, RetryFlow, Source}
+import akka.util.ByteString
+
+import java.time.Clock
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+object AzureStorageStream {
+
+  private[storage] def getObject(storageType: String,
+                                 objectPath: String,
+                                 requestBuilder: RequestBuilder): Source[ByteString, Future[ObjectMetadata]] =
+    Source
+      .fromMaterializer { (mat, attr) =>
+        implicit val system: ActorSystem = mat.system
+        val settings = resolveSettings(attr, system)
+        val request = requestBuilder.createRequest(settings, storageType, objectPath)
+        val objectMetadataMat = Promise[ObjectMetadata]()
+        signAndRequest(request, settings)(mat.system)
+          .map(response => response.withEntity(response.entity.withoutSizeLimit))
+          .mapAsync(parallelism = 1)(entityForSuccess)
+          .flatMapConcat {
+            case (entity, headers) =>
+              objectMetadataMat.success(computeMetaData(headers, entity))
+              entity.dataBytes
+          }
+          .mapError {
+            case e: Throwable =>
+              objectMetadataMat.tryFailure(e)
+              e
+          }
+          .mapMaterializedValue(_ => objectMetadataMat.future)
+      }
+      .mapMaterializedValue(_.flatMap(identity)(ExecutionContext.parasitic))
+
+  private[storage] def getBlobProperties(objectPath: String,
+                                         requestBuilder: GetProperties): Source[Option[ObjectMetadata], NotUsed] =
+    handleRequest(successCode = OK, storageType = BlobType, objectPath = objectPath, requestBuilder = requestBuilder)
+
+  private[storage] def getFileProperties(objectPath: String,
+                                         requestBuilder: GetProperties): Source[Option[ObjectMetadata], NotUsed] =
+    handleRequest(successCode = OK, storageType = FileType, objectPath = objectPath, requestBuilder = requestBuilder)
+
+  private[storage] def deleteBlob(objectPath: String,
+                                  requestBuilder: RequestBuilder): Source[Option[ObjectMetadata], NotUsed] =
+    handleRequest(successCode = Accepted,
+                  storageType = BlobType,
+                  objectPath = objectPath,
+                  requestBuilder = requestBuilder)
+
+  private[storage] def deleteFile(objectPath: String,
+                                  requestBuilder: RequestBuilder): Source[Option[ObjectMetadata], NotUsed] =
+    handleRequest(successCode = Accepted,
+                  storageType = FileType,
+                  objectPath = objectPath,
+                  requestBuilder = requestBuilder)
+
+  private[storage] def putBlob(objectPath: String,
```

Comment:
@sfali one question for large files may be of > 5-10Gb .. S3 has multipartUploadAPI for azure blob how are we going to achieve it ?

splitting in chunks? 

adding some UTs would have been helpful..